common/persistence/execution_manager_test.go

```diff
@@ -0,0 +1,242 @@
+			},
+		},
+		{
+			method: "RangeCompleteReplicationTask",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().RangeCompleteReplicationTask(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "DeleteWorkflowExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().DeleteWorkflowExecution(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "DeleteCurrentWorkflowExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().DeleteCurrentWorkflowExecution(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "GetCurrentExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetCurrentExecution(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "ListCurrentExecutions",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().ListCurrentExecutions(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "IsWorkflowExecutionExists",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().IsWorkflowExecutionExists(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "GetReplicationDLQSize",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetReplicationDLQSize(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "GetReplicationTasks",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetReplicationTasks(gomock.Any(), gomock.Any()).Return(&InternalGetReplicationTasksResponse{
+					Tasks: []*InternalReplicationTaskInfo{
+						{
+							DomainID: "test",
+							TaskID:   1,
+						},
+						{
+							DomainID: "test",
+							TaskID:   2,
+						},
+					},
+					NextPageToken: nil,
+				}, nil)
+			},
+			verifyRes: func(t *testing.T, resValues []reflect.Value) {
+				assert.Len(t, resValues, 2)
+				res := resValues[0].Interface().(*GetReplicationTasksResponse)
+				assert.Len(t, res.Tasks, 2)
+				assert.Equal(t, int64(1), res.Tasks[0].TaskID)
+				assert.Equal(t, int64(2), res.Tasks[1].TaskID)
+				assert.Nil(t, res.NextPageToken)
+				assert.Nil(t, resValues[1].Interface())
+			},
+		},
+	} {
+		t.Run(tc.method, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockedStore := NewMockExecutionStore(ctrl)
+			tc.prepareMocks(mockedStore)
+			manager := NewExecutionManagerImpl(mockedStore, testlogger.New(t), nil)
+			v := reflect.ValueOf(manager)
+			method := v.MethodByName(tc.method)
+			methodType := method.Type()
+			args := methodType.NumIn()
+			var vals []reflect.Value
+			if args > 0 {
+				vals = append(vals, reflect.ValueOf(context.Background()))
+				for i := 1; i < args; i++ {
+					vals = append(vals, reflect.Zero(methodType.In(i)))
+				}
+			}
```

Comment:
This block deserves a comment (or maybe even extraction to a helper function) since it is very low-level, but takes most of the test-body.

Reply:
added a comment